samples/src/main/java/org/corfudb/samples/graph/GraphDB.java

```diff
@@ -0,0 +1,149 @@
+package org.corfudb.samples.graph;
+
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.collections.SMRMap;
+
+import java.util.*;
+
+/**
+ * Created by shriyav on 5/25/17.
+ */
+
+public class GraphDB {
+    private SMRMap<UUID, Node> vertices;
+    CorfuRuntime rt;
+
+    public GraphDB(CorfuRuntime runtime) {
+        vertices = runtime.getObjectsView()
+                .build()
+                .setStreamName("A")     // stream name
+                .setType(SMRMap.class)  // object class backed by this stream
+                .open();                // instantiate the object!
+        rt = runtime;
+    }
+
+    @Override
+    public String toString() {
+        return "This graph has " + getNumNodes() + " vertices.";
+    }
+
+    public SMRMap<UUID, Node> getVertices() { return vertices; }
+
+    public Node getNode(UUID id) { return vertices.get(id); }
+
+    public void addNode(UUID uuid) {
+        vertices.put(uuid, new Node(uuid));
+    }
+
+    public void addNode(UUID uuid, String name) {
+        vertices.put(uuid, new Node(uuid, name));
+    }
+
+    public void addNode(UUID uuid, String name, String type) {
+        if (type.equals("Transport Zone")) {
+            TransportZone tz = new TransportZone(uuid, name);
+            vertices.put(uuid, tz);
+        } else if (type.equals("Transport Node")) {
+            TransportNode tn = new TransportNode(uuid, name);
+            vertices.put(uuid, tn);
+        } else if (type.equals("Logical Switch")) {
+            LogicalSwitch ls = new LogicalSwitch(uuid, name);
+            vertices.put(uuid, ls);
+        } else if (type.equals("Logical Port")) {
+            LogicalPort lp = new LogicalPort(uuid, name);
+            vertices.put(uuid, lp);
+        }
+    }
+
+    public void update(UUID uuid, Map<String, Object> properties) {
+        vertices.get(uuid).getProperties().putAll(properties);
```

Comment:
update should take a node as a parameter and it should be a put on the vertices map. Since the Node type itself isnt an smr object.